app/mailers/reminder_mailer.rb

```diff
@@ -1,36 +1,45 @@
 # frozen_string_literal: true
 
 class ReminderMailer < ApplicationMailer
-  def overdue_email(user, carrier_name, current_location, due_date)
-    @user = user.name.first
-    @carrier_name = carrier_name
-    @current_location = current_location
+  def overdue_email(user, carrier, current_location, due_date)
+    @user = user
+    @user_name = @user.name.first
+    @carrier_name = carrier.name
+    @current_location_name = current_location.name
     @due_date = due_date
     mail(to: @user.email, subject: 'Baby Carrier Overdue For Return')
   end
 
-  def due_today_email(user, carrier_name, current_location)
-    @user = user.name.first
-    @carrier_name = carrier_name
+  def due_today_email(user, carrier, current_location)
+    @user = user
+    @user_name = @user.name.first
+    @carrier_name = carrier.name
+    @manufacturer = carrier.manufacturer
+    @model = carrier.model
     @current_location = current_location
-    mail(to: @user.email, subject: 'Carrier Return Is Now Due')
+    mail(to: @user.email, subject: 'Baby Carrier Due For Return')
   end
 
-  def week_advance_notice_email(user, carrier_name, current_location, due_date)
-    @user = user.name.first
-    @carrier_name = carrier_name
+  def week_advance_notice_email(user, carrier, current_location, due_date)
```

Comment:
Maybe the `due_today_email` and the `week_advance_notice_email` can be combined into one?